### PR TITLE
CameraGimbalUI: check only for gamepads when gimbal is available, ask before capturing it

### PR DESCRIPTION
**Pull request by marvind**
Problem: any gamepad is currently captured by CameraGimbalUI, even when no gimbal is available. This can lead to crashes and undesired behavior (e.g., gamepad should be used in other applications).
Solution: try only to connect to gamepad when gimbal was detected, ask with yes/no dialog whether to connect

### CODE REVIEW

**Review by rickaha — APPROVED** [review]
Nice solution

**Comment by r2avula** [comment]
I am getting a Segmentation fault error due to QMessageBox::question. Perhaps the text is too big for the popup window?

File: userinterface/cameragimbalui.cpp [hunk 1]
```diff
@@ -84,6 +31,60 @@ void CameraGimbalUI::setGimbal(const QSharedPointer<Gimbal> gimbal)
         ui->gimbalStatusLabel->setText("Gimbal found.");
         ui->gimbalControlGroup->setEnabled(true);
         ui->cameraControlGroup->setEnabled(true);
+
+        if (!QGamepadManager::instance()->connectedGamepads().isEmpty() &&
+                QMessageBox::question(this, "CameraGimbalUI: Gamepad detected", "A gamepad was detected. Should it be used for controlling the vehicle's gimbal?") == QMessageBox::Yes) {
```
Comment (r2avula) [inline userinterface/cameragimbalui.cpp:36]: I am getting a Segmentation fault error due to QMessageBox::question. Perhaps the text is too big for the popup window?

**Review by r2avula — CHANGES_REQUESTED** [review]

File: userinterface/cameragimbalui.cpp [hunk 2]
```diff
@@ -84,6 +31,60 @@ void CameraGimbalUI::setGimbal(const QSharedPointer<Gimbal> gimbal)
...
+                    mVehicleConnection->setActuatorOutput(2, 1.0f);
+                valueWas = value;
+            });
+            connect(mGamepad.get(), &QGamepad::buttonR3Changed, [this](bool value) {
+                static bool valueWas = false;
+                if (value && !valueWas)
+                    mVehicleConnection->setActuatorOutput(2, -1.0f);
+                valueWas = value;
+            });
+        } else
```
Comment (r2avula) [inline userinterface/cameragimbalui.cpp:86]:
```suggestion
            ui->gimbalStatusLabel->setText("Gimbal found. Using gamepad \"" + mGamepad->name() + "\" for controlling it.");
            ui->gimbalStatusLabel->setWordWrap(true);
        } else
```

**Review by r2avula — CHANGES_REQUESTED** [review]

File: userinterface/cameragimbalui.cpp [hunk 3]
```diff
@@ -84,6 +30,89 @@ void CameraGimbalUI::setGimbal(const QSharedPointer<Gimbal> gimbal)
...
+            askUseGamepadMessageBox->setText("CameraGimbalUI: Gamepad detected");
+            askUseGamepadMessageBox->setInformativeText("A gamepad was detected: " + mGamepad->name() + ".\nShould it be used for controlling the vehicle's gimbal?");
+            askUseGamepadMessageBox->setIcon(QMessageBox::Information);
+
+            // We do not have any ownership over yesButton -> raw pointer
+            QPushButton* yesButton = askUseGamepadMessageBox->addButton(QMessageBox::Yes);
+            askUseGamepadMessageBox->addButton(QMessageBox::No);
+            askUseGamepadMessageBox->setDefaultButton(QMessageBox::No);
+            
+            QObject::connect(askUseGamepadMessageBox.get(), &QMessageBox::buttonClicked, [this, yesButton](QAbstractButton *button) {
```
[the remark this answers is not in the document]
Comment (marvind) [reply userinterface/cameragimbalui.cpp]: I now found the issue, and hopefully fixed it... Depending on when the gimbal was detected exactly (either during MavsdkVehicleConnection's construction or later), the messagebox was created from a MAVSDK or Qt thread, respectively. Problems arose when created from MAVSDK thread and I now make sure that it is shown from a Qt thread.

**Review by r2avula — APPROVED** [review]
Seems to work as expected now

**Pull request merged**